src/services/social-media.service.ts

```diff
@@ -76,50 +76,36 @@ export class SocialMediaService {
     username: string,
     publicKey: string,
   ): Promise<ExtendedPeople> {
-    let redditUser = null;
-    let foundRedditPost = null;
-
     try {
-      // Fetch data user from reddit api
-      ({data: redditUser} = await this.redditService.getActions(
+      const {data: redditUser} = await this.redditService.getActions(
         `user/${username}/about.json`,
-      ));
-
-      // Fetch post timeline based on reddit username from reddit api
-      ({data: foundRedditPost} = await this.redditService.getActions(
+      );
+      const {data: redditPost} = await this.redditService.getActions(
         `user/${username}/.json?limit=1`,
-      ));
-    } catch {
-      // ignore
-    }
-
-    if (!redditUser) throw new HttpErrors.NotFound('Invalid username');
-    if (!foundRedditPost)
-      throw new HttpErrors.NotFound('Cannot find the speccified post');
-    if (foundRedditPost.children.length === 0)
-      throw new HttpErrors.NotFound('Cannot find the spesified post');
+      );
 
-    // Verify that the publicKey is existing in user reddit
-    const foundRedditPublicKey = foundRedditPost.children[0].data.title
-      .replace(/\n/g, ' ')
-      .split(' ')
-      .find((post: string) => post === publicKey);
+      const found = redditPost.children[0].data.title
```

Comment:
safety null can we use this `const found = redditPost?.children[0]?.data?.title?.replace`